idaes/generic_models/flowsheets/tests/test_methanol_flowsheet.py

```diff
@@ -0,0 +1,193 @@
+"""
+Tests for methanol flowsheet.
+
+"""
+
+import pytest
+
+from idaes.generic_models.flowsheets.methanol_flowsheet import (
+    build_model, set_inputs, initialize_flowsheet, add_costing, report)
+
+from pyomo.environ import (Constraint,
+                           ConcreteModel,
+                           value)
+from pyomo.network import Arc
+from pyomo.environ import TerminationCondition
+from pyomo.util.check_units import assert_units_consistent
+
+from idaes.core import FlowsheetBlock
+from idaes.core.util import get_solver
+from idaes.core.util.model_statistics import degrees_of_freedom
+
+from idaes.generic_models.properties.core.generic.generic_property import \
+    GenericParameterBlock
+from idaes.generic_models.properties.core.generic.generic_reaction import \
+    GenericReactionParameterBlock
+
+from idaes.generic_models.unit_models import (
+    Mixer,
+    Heater,
+    Compressor,
+    Turbine,
+    StoichiometricReactor,
+    Flash)
+
+
+@pytest.fixture(scope='module')
+def model():
+    m = ConcreteModel()
+    m = build_model(m)
+
+    return m
+
+
+@pytest.mark.unit
+def test_build_flowsheet(model):
+    assert isinstance(model.fs, FlowsheetBlock)
+
+    assert isinstance(model.fs.thermo_params, GenericParameterBlock)
+    assert isinstance(model.fs.reaction_params, GenericReactionParameterBlock)
+
+    assert isinstance(model.fs.M101, Mixer)
+    assert isinstance(model.fs.C101, Compressor)
+    assert isinstance(model.fs.H101, Heater)
+    assert isinstance(model.fs.R101, StoichiometricReactor)
+    assert isinstance(model.fs.T101, Turbine)
+    assert isinstance(model.fs.H102, Heater)
+    assert isinstance(model.fs.F101, Flash)
+
+    assert isinstance(model.fs.s02, Arc)
+    assert isinstance(model.fs.s03, Arc)
+    assert isinstance(model.fs.s04, Arc)
+    assert isinstance(model.fs.s05, Arc)
+    assert isinstance(model.fs.s06, Arc)
+    assert isinstance(model.fs.s07, Arc)
+
+    assert degrees_of_freedom(model) == 25
+
+
+@pytest.mark.unit
+def test_set_inputs(model):
+    set_inputs(model)
+
+    assert degrees_of_freedom(model) == 0
+
+
+@pytest.mark.unit
+def test_initialize_flowsheet(model):
+    initialize_flowsheet(model)
+
+    assert degrees_of_freedom(model) == 0
+
+    assert model.fs.M101.outlet.flow_mol[0].value == pytest.approx(954, 1e-3)
+    assert model.fs.C101.outlet.flow_mol[0].value == pytest.approx(954, 1e-3)
+    assert model.fs.H101.outlet.flow_mol[0].value == pytest.approx(954, 1e-3)
+    assert model.fs.R101.outlet.flow_mol[0].value == pytest.approx(478.8, 1e-3)
+    assert model.fs.T101.outlet.flow_mol[0].value == pytest.approx(478.8, 1e-3)
+    assert model.fs.H102.outlet.flow_mol[0].value == pytest.approx(478.8, 1e-3)
+    assert model.fs.F101.vap_outlet.flow_mol[0].expr.value == pytest.approx(
+        336.23, 1e-3)
+    assert model.fs.F101.liq_outlet.flow_mol[0].expr.value == pytest.approx(
+        142.57, 1e-3)
```

Comment:
A check for conservation of mass and energy would be good too. In fact, I don't see any checks for temperature and pressure above at all.